server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetDesiredBalanceAction.java

```diff
@@ -44,4 +45,9 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             new RestRefCountedChunkedToXContentListener<>(restChannel)
         );
     }
+
+    @Override
+    public Set<String> supportedCapabilities() {
+        return Set.of("cluster_balance-node_balance_stats-node_weights_returned");
```

Comment:
Nit: I'd rather not have mixed usage of dashs and underscores. I don't feel it helps readability at all. We use underscores in most other places. So maybe just `desired_balance_node_weights_in_response`?